netbox/dcim/models/device_components.py

```diff
@@ -914,6 +914,10 @@ def save(self, *args, **kwargs):
 
         super().save(*args, **kwargs)
 
+        # Clear any tagged vlans set when mode is tagged-all
+        if self.mode == InterfaceModeChoices.MODE_TAGGED_ALL and self.tagged_vlans.count():
+            self.tagged_vlans.set([])
```

Comment:
IMO we should rely on validation to prevent this, and require the user to explicitly remove all VLANs from the interface. Otherwise, this opens the possibility that all VLAN assignments can be cleared by accidentally changing the 802.1Q mode.

(Also, we can call `self.tagged_vlans.clear()` instead of `set([])`.)